### DISCUSSION TITLE
raw response from plex servers

### DISCUSSION THREAD

**Discussion by seertenedos** [jensenkd/plex-api#67] [Ideas]
One thing that has been making my life a little hard is that currently i can't get to the raw response from plex to see if things are due to things not being mapped, broken mapping or plex does not have or provide that information.  Generally you don't want this information all the time but when looking at new functionality or looking into issues it would be useful.

I can think of 2 ways to do this and currently my feeling is option 2 is the best but i would love a discussion on ideas.
1. Log all the responses out at a debug or trace level.
2. Have a base class for all the object that has a field of type string for the raw response. It could also be a field manually added to objects but a base would make it easier to implement and stop it being missed.  You would also have a setting on the "ClientOptions" class to turn it on and off and by default it is off since it is not needed most of the time.

When i was doing my recent changes to the lib i did notice a lot was not mapped over which was fine as i did not need it at the time but it is important for developers to know what is missing.

**Comment by jensenkd** [comment]
@seertenedos, I've had similar issues.  I was writing out Log.Debug responses for errores, but have just added Log.Info on successes. 

<img width="1461" alt="image" src="https://user-images.githubusercontent.com/170765/170281290-9a353205-f00e-4bc4-a95f-4396ea9f56b8.png">

I'm sure there are lots of gaps that need to be filled.  When I first built this, I really only built it for my use cases.  Hopefully others can help fill in the mapping gaps.

**Reply by seertenedos** [reply]
great thanks

**Comment by jensenkd** [comment]
If you see any missing mappings, you can plug the json into: https://json2csharp.com/

Allows you to quickly add mappings in.
